### PR TITLE
New documentation infrastructure

### PR DESCRIPTION
**Pull request by melissawm**
As discussed in the documentation working group and community meeting, here is a PR with a simplified infrastructure for docs building, including git-lfs for image hosting.

This is a big PR, but it is easier to review looking at the final result than in smaller pieces of code.

This includes:
- Adding git LFS configuration for the repo, with management of gifs, pngs and jpgs;
- Removing jupyter-book and going with a sphinx+myst-nb build;
- Adding the tutorials from napari.github.io to this repo;
- Adding pydata-sphinx-theme while the napari theme is being developed.

To build locally and verify that it works as expected:
- clone this PR
- install napari as usual
- `make docs`

The index for the generated site/documentation will be at `napari/docs/_build/index.html`.

Pending:
- [ ] Since this eliminates `copy-docs.py`, there should be changes in the deployment of the site if this new infrastructure is accepted. This needs to be discussed with the team, especially @codemonkey800 and @kne42 
- [ ] The pydata-sphinx-theme version switcher is not working locally; I can figure this out here or in a subsequent PR.
- [x] The top navbar, wording, and possible re-organization is open to discussion, so this is not necessarily a final version.
- [x] Check for plausible stats

cc @neuromusic @nclack 

Supersedes napari/napari#3842 

## Type of change

- [ ] Bug-fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# References

# How has this been tested?

- [ ] example: the test suite for my feature covers cases x, y, and z
- [ ] example: all tests pass with my change

## Final checklist:
- [x] My PR is the minimum possible work for the desired functionality
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If I included new strings, I have used `trans.` to make them localizable.
      For more information see our [translations guide](https://napari.org/docs/dev/guides/translations.html).

### CODE REVIEW

**Comment by neuromusic** [comment]
re plausible, it looks like this is implemented here: https://github.com/melissawm/napari/blob/new-docs/docs/_templates/base.html, however I believe that we want the url to be `https://plausible.io/js/plausible.outbound-links.js` so that we can track outbound links (see https://plausible.io/docs/script-extensions#all-our-script-extensions)

**Comment by melissawm** [comment]
(Not sure what's happening with the tests - I don't think it is related to these changes, though.)

**Comment by tlambert03** [comment]
I'm curious to hear your thoughts on the removal of jupyter book?  Do you just feel like it's an unnecessary additional layer?

**Comment by tlambert03** [comment]
super enthusiastic about this.

I just pulled and tried to build, but got a dead kernel error twice,  still trying to debug, here's the log


<details>

```
# Sphinx version: 4.4.0
# Python version: 3.9.9 (CPython)
# Docutils version: 0.16 release
# Jinja2 version: 3.0.3
# Last messages:
#   Executing: tutorials/fundamentals/points in: /Users/talley/Dropbox (HMS)/Python/forks/napari/docs/tutorials/fundamentals
#   reading sources... [ 92%] tutorials/fundamentals/shapes
#   Executing: tutorials/fundamentals/shapes in: /Users/talley/Dropbox (HMS)/Python/forks/napari/docs/tutorials/fundamentals
#   reading sources... [ 92%] tutorials/fundamentals/surface
#   Executing: tutorials/fundamentals/surface in: /Users/talley/Dropbox (HMS)/Python/forks/napari/docs/tutorials/fundamentals
#   reading sources... [ 93%] tutorials/fundamentals/tracks
#   reading sources... [ 94%] tutorials/fundamentals/vectors
#   Executing: tutorials/fundamentals/vectors in: /Users/talley/Dropbox (HMS)/Python/forks/napari/docs/tutorials/fundamentals
#   reading sources... [ 94%] tutorials/fundamentals/viewer
#   Executing: tutorials/fundamentals/viewer in: /Users/talley/Dropbox (HMS)/Python/forks/napari/docs/tutorials/fundamentals
# Loaded extensions:
#   sphinx.ext.mathjax (4.4.0) from /Users/talley/miniconda3/envs/napdev/lib/python3.9/site-packages/sphinx/ext/mathjax.py
#   sphinxcontrib.applehelp (1.0.2) from /Users/talley/miniconda3/envs/napdev/lib/python3.9/site-packages/sphinxcontrib/applehelp/__init__.py
#   sphinxcontrib.devhelp (1.0.2) from /Users/talley/miniconda3/envs/napdev/lib/python3.9/site-packages/sphinxcontrib/devhelp/__init__.py
#   sphinxcontrib.htmlhelp (2.0.0) from /Users/talley/miniconda3/envs/napdev/lib/python3.9/site-packages/sphinxcontrib/htmlhelp/__init__.py
#   sphinxcontrib.serializinghtml (1.1.5) from /Users/talley/miniconda3/envs/napdev/lib/python3.9/site-packages/sphinxcontrib/serializinghtml/__init__.py
#   sphinxcontrib.qthelp (1.0.3) from /Users/talley/miniconda3/envs/napdev/lib/python3.9/site-packages/sphinxcontrib/qthelp/__init__.py
#   alabaster (0.7.12) from /Users/talley/miniconda3/envs/napdev/lib/python3.9/site-packages/alabaster/__init__.py
#   sphinx.ext.autodoc.preserve_defaults (1.0) from /Users/talley/miniconda3/envs/napdev/lib/python3.9/site-packages/sphinx/ext/autodoc/preserve_defaults.py
#   sphinx.ext.autodoc.type_comment (4.4.0) from /Users/talley/miniconda3/envs/napdev/lib/python3.9/site-packages/sphinx/ext/autodoc/type_comment.py
#   sphinx.ext.autodoc (4.4.0) from /Users/talley/miniconda3/envs/napdev/lib/python3.9/site-packages/sphinx/ext/autodoc/__init__.py
#   sphinx.ext.napoleon (4.4.0) from /Users/talley/miniconda3/envs/napdev/lib/python3.9/site-packages/sphinx/ext/napoleon/__init__.py
#   sphinx.ext.autosummary (4.4.0) from /Users/talley/miniconda3/envs/napdev/lib/python3.9/site-packages/sphinx/ext/autosummary/__init__.py
#   sphinx.ext.intersphinx (4.4.0) from /Users/talley/miniconda3/envs/napdev/lib/python3.9/site-packages/sphinx/ext/intersphinx.py
#   sphinx_external_toc (0.2.3) from /Users/talley/miniconda3/envs/napdev/lib/python3.9/site-packages/sphinx_external_toc/__init__.py
#   sphinx_tabs.tabs (unknown version) from /Users/talley/miniconda3/envs/napdev/lib/python3.9/site-packages/sphinx_tabs/tabs.py
#   sphinx_togglebutton (0.2.3) from /Users/talley/miniconda3/envs/napdev/lib/python3.9/site-packages/sphinx_togglebutton/__init__.py
#   myst_nb (0.13.1) from /Users/talley/miniconda3/envs/napdev/lib/python3.9/site-packages/myst_nb/__init__.py
#   sphinx_panels (0.6.0) from /Users/talley/miniconda3/envs/napdev/lib/python3.9/site-packages/sphinx_panels/__init__.py
#   sphinx.ext.viewcode (4.4.0) from /Users/talley/miniconda3/envs/napdev/lib/python3.9/site-packages/sphinx/ext/viewcode.py
#   pydata_sphinx_theme (unknown version) from /Users/talley/miniconda3/envs/napdev/lib/python3.9/site-packages/pydata_sphinx_theme/__init__.py
Traceback (most recent call last):
  File "/Users/talley/miniconda3/envs/napdev/lib/python3.9/site-packages/nbclient/client.py", line 618, in _async_poll_for_reply
    msg = await ensure_async(self.kc.shell_channel.get_msg(timeout=new_timeout))
  File "/Users/talley/miniconda3/envs/napdev/lib/python3.9/site-packages/nbclient/util.py", line 96, in ensure_async
    result = await obj
  File "/Users/talley/miniconda3/envs/napdev/lib/python3.9/site-packages/jupyter_client/channels.py", line 224, in get_msg
    ready = await self.socket.poll(timeout)
asyncio.exceptions.CancelledError

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/talley/miniconda3/envs/napdev/lib/python3.9/site-packages/nbclient/client.py", line 830, in async_execute_cell
    exec_reply = await self.task_poll_for_reply
asyncio.exceptions.CancelledError

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/talley/miniconda3/envs/napdev/lib/python3.9/site-packages/sphinx/cmd/build.py", line 284, in build_main
    app.build(args.force_all, filenames)
  File "/Users/talley/miniconda3/envs/napdev/lib/python3.9/site-packages/sphinx/application.py", line 337, in build
    self.builder.build_update()
  File "/Users/talley/miniconda3/envs/napdev/lib/python3.9/site-packages/sphinx/builders/__init__.py", line 294, in build_update
    self.build(to_build,
  File "/Users/talley/miniconda3/envs/napdev/lib/python3.9/site-packages/sphinx/builders/__init__.py", line 308, in build
    updated_docnames = set(self.read())
  File "/Users/talley/miniconda3/envs/napdev/lib/python3.9/site-packages/sphinx/builders/__init__.py", line 415, in read
    self._read_serial(docnames)
  File "/Users/talley/miniconda3/envs/napdev/lib/python3.9/site-packages/sphinx/builders/__init__.py", line 436, in _read_serial
    self.read_doc(docname)
  File "/Users/talley/miniconda3/envs/napdev/lib/python3.9/site-packages/sphinx/builders/__init__.py", line 476, in read_doc
    doctree = read_doc(self.app, self.env, self.env.doc2path(docname))
  File "/Users/talley/miniconda3/envs/napdev/lib/python3.9/site-packages/sphinx/io.py", line 189, in read_doc
    pub.publish()
  File "/Users/talley/miniconda3/envs/napdev/lib/python3.9/site-packages/docutils/core.py", line 217, in publish
    self.document = self.reader.read(self.source, self.parser,
  File "/Users/talley/miniconda3/envs/napdev/lib/python3.9/site-packages/sphinx/io.py", line 109, in read
    self.parse()
  File "/Users/talley/miniconda3/envs/napdev/lib/python3.9/site-packages/docutils/readers/__init__.py", line 77, in parse
    self.parser.parse(self.input, document)
  File "/Users/talley/miniconda3/envs/napdev/lib/python3.9/site-packages/myst_nb/parser.py", line 67, in parse
    ntbk = generate_notebook_outputs(
  File "/Users/talley/miniconda3/envs/napdev/lib/python3.9/site-packages/myst_nb/execution.py", line 137, in generate_notebook_outputs
    result = single_nb_execution(
  File "/Users/talley/miniconda3/envs/napdev/lib/python3.9/site-packages/jupyter_cache/executors/utils.py", line 51, in single_nb_execution
    executenb(
  File "/Users/talley/miniconda3/envs/napdev/lib/python3.9/site-packages/nbclient/client.py", line 1085, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
  File "/Users/talley/miniconda3/envs/napdev/lib/python3.9/site-packages/nbclient/util.py", line 84, in wrapped
    return just_run(coro(*args, **kwargs))
  File "/Users/talley/miniconda3/envs/napdev/lib/python3.9/site-packages/nbclient/util.py", line 62, in just_run
    return loop.run_until_complete(coro)
  File "/Users/talley/miniconda3/envs/napdev/lib/python3.9/asyncio/base_events.py", line 642, in run_until_complete
    return future.result()
  File "/Users/talley/miniconda3/envs/napdev/lib/python3.9/site-packages/nbclient/client.py", line 551, in async_execute
    await self.async_execute_cell(
  File "/Users/talley/miniconda3/envs/napdev/lib/python3.9/site-packages/nbclient/client.py", line 834, in async_execute_cell
    raise DeadKernelError("Kernel died")
nbclient.exceptions.DeadKernelError: Kernel died

```

</details>

**Comment by melissawm** [comment]
I think I might have to pin some of the dependencies... I'll debug and try to make it more robust. Thanks for flagging!

**Comment by melissawm** [comment]
> I'm curious to hear your thoughts on the removal of jupyter book? Do you just feel like it's an unnecessary additional layer?

Mostly, yes. Because we want to have the api docs with autosummary etc together with the notebooks, it seems like we would need some extra flexibility that jupyterbook does not handle that well at this time - as I understand it that is not what is was built for either. Also jupyterbook is a thin layer over sphinx+mystnb, which means we don't lose much by moving away from it.

**Comment by melissawm** [comment]
Ok - on a clean conda environment, I cannot reproduce the dead kernel - tried both mac and linux... 

Here's my environment:
<details>

```
(napari-dev) ~ conda list
# packages in environment at /Users/mwebermendonca/miniconda3/envs/napari-dev:
#
# Name                    Version                   Build  Channel
alabaster                 0.7.12                   pypi_0    pypi
anyio                     3.5.0                    pypi_0    pypi
appdirs                   1.4.4                    pypi_0    pypi
appnope                   0.1.2                    pypi_0    pypi
argon2-cffi               21.3.0                   pypi_0    pypi
argon2-cffi-bindings      21.2.0                   pypi_0    pypi
asttokens                 2.0.5                    pypi_0    pypi
attrs                     21.4.0                   pypi_0    pypi
babel                     2.9.1                    pypi_0    pypi
backcall                  0.2.0                    pypi_0    pypi
beautifulsoup4            4.10.0                   pypi_0    pypi
black                     22.1.0                   pypi_0    pypi
bleach                    4.1.0                    pypi_0    pypi
bzip2                     1.0.8                h0d85af4_4    conda-forge
ca-certificates           2021.10.8            h033912b_0    conda-forge
cachey                    0.2.1                    pypi_0    pypi
certifi                   2021.10.8                pypi_0    pypi
cffi                      1.15.0                   pypi_0    pypi
charset-normalizer        2.0.11                   pypi_0    pypi
click                     7.1.2                    pypi_0    pypi
cloudpickle               2.0.0                    pypi_0    pypi
colorama                  0.4.4                    pypi_0    pypi
dask                      2022.1.1                 pypi_0    pypi
debugpy                   1.5.1                    pypi_0    pypi
decorator                 5.1.1                    pypi_0    pypi
defusedxml                0.7.1                    pypi_0    pypi
docstring-parser          0.13                     pypi_0    pypi
docutils                  0.16                     pypi_0    pypi
entrypoints               0.4                      pypi_0    pypi
executing                 0.8.2                    pypi_0    pypi
freetype-py               2.2.0                    pypi_0    pypi
fsspec                    2022.1.0                 pypi_0    pypi
gitdb                     4.0.9                    pypi_0    pypi
gitpython                 3.1.26                   pypi_0    pypi
greenlet                  1.1.2                    pypi_0    pypi
heapdict                  1.0.1                    pypi_0    pypi
hsluv                     5.0.2                    pypi_0    pypi
idna                      3.3                      pypi_0    pypi
imageio                   2.14.1                   pypi_0    pypi
imagesize                 1.3.0                    pypi_0    pypi
importlib-metadata        4.10.1                   pypi_0    pypi
intervaltree              3.1.0                    pypi_0    pypi
ipykernel                 6.8.0                    pypi_0    pypi
ipython                   7.31.1                   pypi_0    pypi
ipython-genutils          0.2.0                    pypi_0    pypi
ipywidgets                7.6.5                    pypi_0    pypi
jedi                      0.18.1                   pypi_0    pypi
jinja2                    3.0.3                    pypi_0    pypi
jsonschema                4.4.0                    pypi_0    pypi
jupyter-cache             0.4.3                    pypi_0    pypi
jupyter-client            7.1.2                    pypi_0    pypi
jupyter-core              4.9.1                    pypi_0    pypi
jupyter-server            1.13.4                   pypi_0    pypi
jupyter-server-mathjax    0.2.3                    pypi_0    pypi
jupyter-sphinx            0.3.2                    pypi_0    pypi
jupyterlab-pygments       0.1.2                    pypi_0    pypi
jupyterlab-widgets        1.0.2                    pypi_0    pypi
kiwisolver                1.3.2                    pypi_0    pypi
libffi                    3.4.2                h0d85af4_5    conda-forge
libzlib                   1.2.11            h9173be1_1013    conda-forge
locket                    0.2.1                    pypi_0    pypi
magicgui                  0.3.4                    pypi_0    pypi
markdown-it-py            1.1.0                    pypi_0    pypi
markupsafe                2.0.1                    pypi_0    pypi
matplotlib-inline         0.1.3                    pypi_0    pypi
mdit-py-plugins           0.2.8                    pypi_0    pypi
mistune                   0.8.4                    pypi_0    pypi
mypy-extensions           0.4.3                    pypi_0    pypi
myst-nb                   0.13.1                   pypi_0    pypi
myst-parser               0.15.2                   pypi_0    pypi
napari                    0.4.11rc2.dev360+g36e9b8d5           dev_0    <develop>
napari-console            0.0.4                    pypi_0    pypi
napari-plugin-engine      0.2.0                    pypi_0    pypi
napari-svg                0.1.6                    pypi_0    pypi
nbclient                  0.5.10                   pypi_0    pypi
nbconvert                 6.4.1                    pypi_0    pypi
nbdime                    3.1.1                    pypi_0    pypi
nbformat                  5.1.3                    pypi_0    pypi
ncurses                   6.3                  he49afe7_0    conda-forge
nest-asyncio              1.5.4                    pypi_0    pypi
networkx                  2.6.3                    pypi_0    pypi
notebook                  6.4.8                    pypi_0    pypi
npe2                      0.1.2                    pypi_0    pypi
numpy                     1.22.2                   pypi_0    pypi
numpydoc                  1.2                      pypi_0    pypi
openssl                   3.0.0                h0d85af4_2    conda-forge
packaging                 21.3                     pypi_0    pypi
pandas                    1.4.0                    pypi_0    pypi
pandocfilters             1.5.0                    pypi_0    pypi
parso                     0.8.3                    pypi_0    pypi
partd                     1.2.0                    pypi_0    pypi
pathspec                  0.9.0                    pypi_0    pypi
pexpect                   4.8.0                    pypi_0    pypi
pickleshare               0.7.5                    pypi_0    pypi
pillow                    9.0.1                    pypi_0    pypi
pint                      0.18                     pypi_0    pypi
pip                       22.0.3             pyhd8ed1ab_0    conda-forge
platformdirs              2.4.1                    pypi_0    pypi
pooch                     1.6.0                    pypi_0    pypi
prometheus-client         0.13.1                   pypi_0    pypi
prompt-toolkit            3.0.26                   pypi_0    pypi
psutil                    5.9.0                    pypi_0    pypi
psygnal                   0.2.0                    pypi_0    pypi
ptyprocess                0.7.0                    pypi_0    pypi
pure-eval                 0.2.2                    pypi_0    pypi
pycparser                 2.21                     pypi_0    pypi
pydantic                  1.9.0                    pypi_0    pypi
pydata-sphinx-theme       0.8.0                    pypi_0    pypi
pygments                  2.11.2                   pypi_0    pypi
pyopengl                  3.1.5                    pypi_0    pypi
pyparsing                 3.0.7                    pypi_0    pypi
pyqt5                     5.15.6                   pypi_0    pypi
pyqt5-qt5                 5.15.2                   pypi_0    pypi
pyqt5-sip                 12.9.1                   pypi_0    pypi
pyrsistent                0.18.1                   pypi_0    pypi
python                    3.9.10          hea1dfa3_2_cpython    conda-forge
python-dateutil           2.8.2                    pypi_0    pypi
python_abi                3.9                      2_cp39    conda-forge
pytomlpp                  1.0.10                   pypi_0    pypi
pytz                      2021.3                   pypi_0    pypi
pywavelets                1.2.0                    pypi_0    pypi
pyyaml                    6.0                      pypi_0    pypi
pyzmq                     22.3.0                   pypi_0    pypi
qtconsole                 5.2.2                    pypi_0    pypi
qtpy                      2.0.1                    pypi_0    pypi
readline                  8.1                  h05e3726_0    conda-forge
requests                  2.27.1                   pypi_0    pypi
scikit-image              0.19.1                   pypi_0    pypi
scipy                     1.7.3                    pypi_0    pypi
send2trash                1.8.0                    pypi_0    pypi
setuptools                60.7.1           py39h6e9494a_0    conda-forge
six                       1.16.0                   pypi_0    pypi
smmap                     5.0.0                    pypi_0    pypi
sniffio                   1.2.0                    pypi_0    pypi
snowballstemmer           2.2.0                    pypi_0    pypi
sortedcontainers          2.4.0                    pypi_0    pypi
soupsieve                 2.3.1                    pypi_0    pypi
sphinx                    4.4.0                    pypi_0    pypi
sphinx-autodoc-typehints  1.12.0                   pypi_0    pypi
sphinx-external-toc       0.2.3                    pypi_0    pypi
sphinx-panels             0.6.0                    pypi_0    pypi
sphinx-tabs               3.2.0                    pypi_0    pypi
sphinx-togglebutton       0.2.3                    pypi_0    pypi
sphinxcontrib-applehelp   1.0.2                    pypi_0    pypi
sphinxcontrib-devhelp     1.0.2                    pypi_0    pypi
sphinxcontrib-htmlhelp    2.0.0                    pypi_0    pypi
sphinxcontrib-jsmath      1.0.1                    pypi_0    pypi
sphinxcontrib-qthelp      1.0.3                    pypi_0    pypi
sphinxcontrib-serializinghtml 1.1.5                    pypi_0    pypi
sqlalchemy                1.4.31                   pypi_0    pypi
sqlite                    3.37.0               h23a322b_0    conda-forge
stack-data                0.1.4                    pypi_0    pypi
superqt                   0.2.5.post1              pypi_0    pypi
terminado                 0.13.1                   pypi_0    pypi
testpath                  0.5.0                    pypi_0    pypi
tifffile                  2022.2.2                 pypi_0    pypi
tk                        8.6.11               h5dbffcc_1    conda-forge
tomli                     2.0.0                    pypi_0    pypi
toolz                     0.11.2                   pypi_0    pypi
tornado                   6.1                      pypi_0    pypi
tqdm                      4.62.3                   pypi_0    pypi
traitlets                 5.1.1                    pypi_0    pypi
typer                     0.4.0                    pypi_0    pypi
typing-extensions         4.0.1                    pypi_0    pypi
tzdata                    2021e                he74cb21_0    conda-forge
urllib3                   1.26.8                   pypi_0    pypi
vispy                     0.9.6                    pypi_0    pypi
wcwidth                   0.2.5                    pypi_0    pypi
webencodings              0.5.1                    pypi_0    pypi
websocket-client          1.2.3                    pypi_0    pypi
wheel                     0.37.1             pyhd8ed1ab_0    conda-forge
widgetsnbextension        3.5.2                    pypi_0    pypi
wrapt                     1.13.3                   pypi_0    pypi
xz                        5.2.5                haf1e3a3_1    conda-forge
zipp                      3.7.0                    pypi_0    pypi
zlib                      1.2.11            h9173be1_1013    conda-forge
```

</details>

**Comment by tlambert03** [comment]
> on a clean conda environment, I cannot reproduce the dead kernel 

me too!  sorry you had to check that for me.  just tested on both py39 and py310 and both worked with:
```
conda create -n docbuild python
conda activate docbuild
pip install -e .[dev]
make docs
```

must have been a plugin or something else in the other env.

**Comment by jni** [comment]
Thanks @melissawm!

One Q: I would distinguish between images that are for the docs (screenshots and animations; which, btw, could you add webm, webp, mp4, and mov to the list of lfs-managed files?), and images that are for the application's functionality, specifically, the logo (shown on the splash screen and in the background of an empty viewer), and the loading.gif animation. These files should be managed by plain git, because you will otherwise not be able to get a functional viewer without git-lfs. Does git-lfs allow this kind of exception?

**Comment by melissawm** [comment]
> Does git-lfs allow this kind of exception?

Yes - check out my latest commit: I excluded the logo from the list of files to be tracked by git-lfs. Let me know if there are other specific folders you feel should be removed and I can do that. Ideally those would all live in the same folder so we could remove them in one go instead of removing each file path manually in `.gitattributes`.

**Comment by melissawm** [comment]
Folks, in the latest docs working group and community meetings we discussed the content to be added to the top nav bar. I'm adding Tom (@tdimino) to the conversation here. Here's what's currently in the nav bar:

- Usage
- Plugins
- Community
- API Reference
- napari hub (this could be converted into an icon just like the github repo has, but it seems to me that we want to highlight this a bit more, so I left it here.

Screen Shot 2022-02-07 at 1 20 08 PM

Feedback or suggestions on wording or organization is appreciated.

**Comment by tdimino** [comment]
> Folks, in the latest docs working group and community meetings we discussed the content to be added to the top nav bar. I'm adding Tom (@tdimino) to the conversation here. Here's what's currently in the nav bar:
> 
> * Usage
> * Plugins
> * Community
> * API Reference
> * napari hub (this could be converted into an icon just like the github repo has, but it seems to me that we want to highlight this a bit more, so I left it here.
> 
> Screen Shot 2022-02-07 at 1 20 08 PM
> 
> Feedback or suggestions on wording or organization is appreciated.

I agree completely with Melissa's framing of this top bar nav. I think it solves most of the issues I'd flagged in my own audit of napari.org--namely the lack of grouping between links. 

In the addition to the 5 top level links that Melissa's proposed, I'm curious on everyone's thoughts about elevating "napari workshops" so that it's visible and accessible from the ground level of the nav. At the moment visitors can only access the "napari workshops" page from under "Tutorials". 

Much like other data science sites with workshops, I'm imagining that said "napari workshop" page will evolve to include a section of "upcoming workshops" in addition to its current listing of past workshops.

File: .gitattributes
Comment (sofroniewn) [inline .gitattributes]: We actually need to include everything inside the [napari/resources/](https://github.com/napari/napari/tree/main/napari/resources) folder. Including the `logo.png` and `loading.gif` copies that are there. The `docs/images/logo.png` that you have there shouldn't need to be excluded.

Hopefully fixing this will fix our tests are ultimately they are erroring with

```pytb
  E       ValueError: Could not find a backend to open `/home/runner/work/napari/napari/napari/resources/logo.png`` with iomode `ri`.
```

**Comment by jni** [comment]
I could see Community > Workshops and Using > Workshops both making sense. I don't think it needs to be decided in this PR but I agree that it would be nice to have them accessible from the home page.

File: .gitattributes
Comment (neuromusic) [reply .gitattributes]: @sofroniewn is the separation of concerns such that all image files in `napari/resources` SHOULD NOT be under LFS and all image files in `docs` SHOULD?

Comment (sofroniewn) [reply .gitattributes]: yes - to the best of my understanding - all image files in napari/resources are used by the Qt application (i.e. splash screen) and as such need to be included directly in the repo so our CI passes

all image files inside docs are only needed for docs and can live only in LFS

Comment (melissawm) [reply .gitattributes]: That sounds good - I'll update

**Comment by sofroniewn** [comment]
Playing around with this now. I love it. A couple thoughts - that don't need to be addressed in this PR but I want to capture

- The napari logo in the top left doesn't load for me
- I'd move `napari workshop` out of `napari tutorials` and under `usage` (so not the top nav, but also not at the third level)
- `Single cell tracking with napari` and `Annotating segmentation with text and bounding boxes` appear both in the left nav at the level of `napari tutorials` and but are also under `Tracking` and `Segmentation`. I guess these will be triple nested when that works
- Can we put `Annotating videos with napari` under `Annotation`
- Can we put `Using Dask and napari to process & view large datasets` under `Processing`
- We will then end up with `napari tutorials` -> `Segmentation`, `Tracking`, `Annotation`, and `Processing`. Future tutorials can add to those sections, or create new sections.
- In community can we put `Handling Code of Conduct Reports` under `Code of Conduct` somehow, I don't think it needs to also be it's own top line item, but can link out

**Review by sofroniewn — APPROVED** [review]
I'm approving this now. If someone else can confirm the logo loads that would be great. If people like my other suggestions here https://github.com/napari/napari/pull/4047#issuecomment-1032989594 then we could make them, but if they are controversial/ require discussion then they should wait.

I'd like to get approvals from @jni and @tlambert03 on this PR and then coordinate with @neuromusic and @codemonkey800 on the exact merge time, which could be as soon as today or if not tomorrow I think!! 🚀

**Comment by neuromusic** [comment]
@sofroniewn I don't think that the reorg suggestions are controversial, but this PR is already pretty big and I'd opt to handle the content reorg in a separate content-focused PR

**Comment by sofroniewn** [comment]
> @sofroniewn I don't think that the reorg suggestions are controversial, but this PR is already pretty big and I'd opt to handle the content reorg in a separate content-focused PR

Totally fine. As long as logo is fine then, this is ready by me

**Comment by jni** [comment]
Did plausible get handled already? Do we want to handle it before merge? Just checking through the list in the PR description, and Justin's comment. I don't see any commit dealing with it...?

**Comment by jni** [comment]
One minor comment: I noticed that `make docs` uses `pip install -qr docs/requirements.txt`. Should we instead use `pip install .[docs]`? ie it would be nice to have a single source of truth for our dependencies, and it would be nice if that source was the setup.cfg. But I understand that this is actually different (we install the dependencies *as well as* napari), but maybe that's better?

Overall I'm loving the workflow though... :smiley: Build still running...

**Comment by neuromusic** [comment]
re: plausible, my comment is a suggested improvement and need not land in this PR. added an issue to track it here: https://github.com/napari/docs/issues/362

File: docs/_static/version_switcher.json
Comment (tlambert03) [inline docs/_static/version_switcher.json:1]: this seems like a test right (based on the versions)?  would be nice to remove any in-process stuff before merge

**Review by tlambert03 — APPROVED** [review]
This is just awesome @melissawm.  Looks great, feels great.  I'm more than happy to merge this as is and begin smaller, incremental improvements from here.  Thank you!

> Should we instead use pip install .[docs]

I'm +1 on this idea too

**Comment by DragaDoncila** [comment]
@melissawm I just pulled this down to try it and it works beautifully ❤️ . However, I'm seeing the same thing as @sofroniewn  which is the logo missing - this is on a brand new env with `git lfs` installed, if that makes any difference.

image

**Comment by melissawm** [comment]
Thanks for the feedback folks - I'll investigate the logo issue and the version switcher.

**Comment by melissawm** [comment]
For the version switcher: I was asking around (see, for example https://github.com/scipy/scipy/pull/15380) and it looks like at this point the best plan is to wait for the next release of the pydata-sphinx-theme if we want to use their machinery. It looks like it is experimental right now and has a few limitations which will be solved later - among those is the possibility of styling it in an easier way to have it match the rest of the color scheme for the site (see https://github.com/pydata/pydata-sphinx-theme/issues/566). So If that's ok with you folks, I'll leave the `version-switcher.json` file in but remove the actual switcher in the nav-bar.

**Comment by sofroniewn** [comment]
> So If that's ok with you folks, I'll leave the version-switcher.json file in but remove the actual switcher in the nav-bar.

Totally fine with me

**Review by codemonkey800 — COMMENTED** [review]
Didn't review all of the docs files, but the infra LGTM. @sofroniewn I just noticed that the workflows for deploying to `napari/napari.github.io` are actually one of the things still pending:

> Since this eliminates copy-docs.py, there should be changes in the deployment of the site if this new infrastructure is accepted.

@melissawm to get this working, I think the main thing we'll need to change is update the existing workflow to push to `gh-pages` instead of main:
https://github.com/napari/napari/blob/9db0e37c17b1ea541d10dd1a7baddfe30a0d5c9e/.github/workflows/make_docs.yml#L45-L55

One thing though is this action clones the repo and then pushes a commit. I think there's a way to do this in one go using something like https://github.com/cpina/github-action-push-to-another-repository

Exciting to see it all come together 🎉

**Comment by codemonkey800** [comment]
@melissawm @sofroniewn opened a PR to disable the copy docs workflow
https://github.com/melissawm/napari/pull/1

@sofroniewn @tlambert03 I'm not actually familiar enough with the `build` job. I think this already uses `make docs`, should we comment it out too or leave it in? https://github.com/melissawm/napari/pull/1#discussion_r804266642

**Comment by sofroniewn** [comment]
Thoughts here @melissawm are based on discussion in todays community meeting where we said we want to try and get this PR merged asap without it actually updating website and then do a followup PR where we actually update the site. Having a 120 file modifying PR open a long time just makes me nervous so we want to get it in 🤣 🤣

**Comment by codemonkey800** [comment]
In yesterday's sync, @jni and I talked a briefly about using the napari sphinx theme deploying the docs so that there isn't a gap in the look-and-feel while developing the theme. Since we're working with a fork, all configurations will work the same. I think this makes sense, but curious what folks think
cc: @melissawm @sofroniewn @tlambert03 @neuromusic 

The theme is located [here](https://github.com/napari/napari-sphinx-theme/tree/napari-sphinx-theme-v0) and is pip-installable, only thing we need to do is publish it to PyPi

**Comment by sofroniewn** [comment]
> The theme is located [here](https://github.com/napari/napari-sphinx-theme/tree/napari-sphinx-theme-v0) and is pip-installable, only thing we need to do is publish it to PyPi


I'd love us to switch over to using it right away, but i'd like to merge this PR first. It's got test fails which have been fixed on main.

Maybe once @melissawm has taken one more pass through this and gives me the green light i will merge this and then we can press on with further improvements / switching it on in other PRs. Thanks!!

**Review by jni — APPROVED** [review]
I've merged in main and removed the copy-docs action. I've also substituted in two new actions, build-and-push-artifact to build docs on each PR and upload a corresponding artifact, and build-and-deploy, *with the deploy step commented out*, that gets built on pushes to main or tag pushes.

The build docs use caching for git-lfs based on the pattern described here:

https://github.com/actions/checkout/issues/165#issuecomment-657673315

So :crossed_fingers: that it works so that we can build docs in PRs without burning through bandwidth :joy: If it turns out to not work, we can always build docs on PRs without the images — thanks @melissawm for building infrastructure that works even when the images are missing!

**Comment by jni** [comment]
I'm seeing a lot of warnings in the CI docs build... Any ideas? e.g. https://github.com/napari/napari/runs/5211449270?check_suite_focus=true#step:6:1039

BUT, they seem to be building anyway — we can still merge and improve the build over time.

**Comment by sofroniewn** [comment]
Looks great! Build process worked - deploy code is commented out so merging this should not update website, but it will be ready to go when we are ready, thanks @jni. Also thanks so much @melissawm for all your hard work here, its really been an incredible effort.

**Comment by melissawm** [comment]
Hi folks - sorry I have been super busy with other projects, very happy to see this go through! Yay :tada:

> I'm seeing a lot of warnings in the CI docs build... Any ideas? e.g. [napari/napari/runs/5211449270?check_suite_focus=true#step:6:1039](https://github.com/napari/napari/runs/5211449270?check_suite_focus=true#step:6:1039)
> 
> BUT, they seem to be building anyway — we can still merge and improve the build over time.

Yes, these were always there I'm afraid - they are not new. I actually managed to fix some of the errors in previous PRs but not all.

Thank you so much @codemonkey800 for the heavy lifting on the theme and deployment here, please let me know if there are any pending actions for this PR. I'll coordinate with the team for next steps :)

**Pull request merged**
